drivers/platform/maxim/max32670/maxim_irq.c

```diff
@@ -0,0 +1,581 @@
+			ret = no_os_list_init(&_events[callback_desc->event].actions,
+					      NO_OS_LIST_PRIORITY_LIST,
+					      irq_action_cmp);
+			if (ret)
+				return ret;
+		}
+
+		ret = no_os_list_read_find(_events[callback_desc->event].actions,
+					   (void **)&action,
+					   &action_key);
+		/*
+		 * If an action with the same irq_id as the function parameter does not exists, insert a new one,
+		 * otherwise update
+		 */
+		if (ret) {
+			action = calloc(1, sizeof(*action));
+			if (!action)
+				return -ENOMEM;
+
+			action->irq_id = irq_id;
+			action->handle = callback_desc->handle;
+			action->callback = callback_desc->callback;
+			action->ctx = callback_desc->ctx;
+
+			ret = no_os_list_add_last(_events[callback_desc->event].actions, action);
+			if (ret)
+				goto free_action;
+		} else {
+			action->irq_id = irq_id;
+			action->handle = callback_desc->handle;
+			action->callback = callback_desc->callback;
+			action->ctx = callback_desc->ctx;
+		}
+
+		break;
+
+	case NO_OS_RTC_IRQ:
+		if (_events[NO_OS_EVT_RTC].actions == NULL) {
+			ret = no_os_list_init(&_events[NO_OS_EVT_RTC].actions, NO_OS_LIST_PRIORITY_LIST,
+					      irq_action_cmp);
+			if (ret)
+				return ret;
+		}
+
+		/*
+		 * This is a special case for RTC on Maxim platform. Since there is only 1 RTC peripheral, there should
+		 * be only 1 registered callback at a time.
+		 */
+		ret = no_os_list_read_first(_events[NO_OS_EVT_RTC].actions, (void **)&action);
+		if (ret) {
+			action = calloc(1, sizeof(*action));
+			if (!action)
+				return -ENOMEM;
+
+			action->irq_id = irq_id;
+			action->handle = callback_desc->handle;
+			action->callback = callback_desc->callback;
+			action->ctx = callback_desc->ctx;
+
+			ret = no_os_list_add_first(_events[NO_OS_EVT_RTC].actions, action);
+			if (ret)
+				goto free_action;
+
+		} else {
+			action->irq_id = irq_id;
+			action->handle = callback_desc->handle;
+			action->callback = callback_desc->callback;
+			action->ctx = callback_desc->ctx;
+		}
+
+		ret = MXC_RTC_EnableInt(MXC_RTC_INT_EN_LONG);
+		if (ret)
+			return -EBUSY;
+
+		break;
+
+	case NO_OS_TIM_IRQ:
+		if (_events[NO_OS_EVT_RTC].actions == NULL) {
+			ret = no_os_list_init(&_events[NO_OS_EVT_TIM_ELAPSED].actions,
+					      NO_OS_LIST_PRIORITY_LIST,
+					      irq_action_cmp);
+			if (ret)
+				return ret;
+		}
+
+		ret = no_os_list_read_first(_events[NO_OS_EVT_TIM_ELAPSED].actions,
+					    (void **)&action);
+		if (ret) {
+			action = calloc(1, sizeof(*action));
+			if (!action)
+				return -ENOMEM;
+
+			action->irq_id = irq_id;
+			action->handle = callback_desc->handle;
+			action->callback = callback_desc->callback;
+			action->ctx = callback_desc->ctx;
+
+			ret = no_os_list_add_first(_events[NO_OS_EVT_TIM_ELAPSED].actions, action);
+			if (ret)
+				goto free_action;
+
+		} else {
+			action->irq_id = irq_id;
+			action->handle = callback_desc->handle;
+			action->callback = callback_desc->callback;
+			action->ctx = callback_desc->ctx;
+		}
+
+		break;
+
```

Comment:
nit: extra line before and after `break;` check the entire code for these occurrences.

Reply:
Will remove extra space before and after break